File: day-05/python/part1.py
#!/usr/bin/env python3

import sys

data = [x.replace('F', '0').replace('B', '1').replace('L', '0').replace('R', '1') for x in open(sys.argv[1], "r").read().split("\n")]

sID = []

for line in data:
    fRow = int(line[:7], 2)
    fCol = int(line[-3:], 2)

    sID.append(fRow * 8 + fCol)

print(max(sID))

File: day-05/nim/part1.nim
from os import paramStr
import strutils, sugar, algorithm

let data = collect(newSeq):
  for x in lines(paramStr(1)):
    x.multiReplace(("F", "0"), ("B", "1"), ("L", "0"), ("R", "1"))
var sID = newSeq[int](0)

for line in data:
  var
    fRow = parseBinInt(line[0..^4])
    fCol = parseBinInt(line[^3..^1])

  sID.add(fRow * 8 + fCol)

sort(sID)
echo sID[^1]
